src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -209,57 +226,37 @@ impl PrivilegeAccess {
-                }
-            }
+        // translate the db_id and table_id to db_name and table_name, used on
+        // skipping the privilege check on system tables, and on the error message.
+        let (db_name, table_name) = match object {
+            GrantObject::Database(_, db_name) => (db_name.to_lowercase(), "".to_string()),
             GrantObject::Table(_, db_name, table_name) => {
-                let db_name = db_name.to_lowercase();
-                let table_name = table_name.to_lowercase();
-                if (db_name.to_lowercase() == "information_schema"
-                    || (db_name == "system" && table_name == "one"))
-                    && privileges.contains(&UserPrivilegeType::Select)
-                    && len == 1
-                {
-                    return Ok(());
-                }
+                (db_name.to_lowercase(), table_name.to_lowercase())
             }
             GrantObject::DatabaseById(catalog_name, db_id) => {
                 let catalog = self.ctx.get_catalog(catalog_name).await?;
-                db_name = catalog.get_db_name_by_id(*db_id).await?.to_lowercase();
-                if db_name == "information_schema"
-                    && privileges.contains(&UserPrivilegeType::Select)
-                    && len == 1
-                {
-                    return Ok(());
-                }
+                let db_name = catalog.get_db_name_by_id(*db_id).await?;
+                (db_name.to_lowercase(), "".to_string())
             }
             GrantObject::TableById(catalog_name, db_id, table_id) => {
                 let catalog = self.ctx.get_catalog(catalog_name).await?;
-                db_name = catalog.get_db_name_by_id(*db_id).await?.to_lowercase();
-                table_name = catalog
-                    .get_table_name_by_id(*table_id)
-                    .await?
-                    .to_lowercase();
-                if (db_name == "information_schema" || (db_name == "system" && table_name == "one"))
-                    && privileges.contains(&UserPrivilegeType::Select)
-                    && len == 1
-                {
-                    return Ok(());
-                }
+                let db_name = catalog.get_db_name_by_id(*db_id).await?;
+                let table_name = catalog.get_table_name_by_id(*table_id).await?;
+                (db_name.to_lowercase(), table_name.to_lowercase())
             }
-            GrantObject::Global | GrantObject::Stage(_) | GrantObject::UDF(_) => {}
+            _ => ("".to_string(), "".to_string()),
+        };
+
+        // skip checking the privilege on system tables.
+        if ((db_name == "system" && SYSTEM_TABLES_ALLOW_LIST.iter().any(|x| x == &table_name))
+            || db_name == "information_schema")
+            && privileges == [UserPrivilegeType::Select]
+        {
+            return Ok(());
         }
+
+        // TODO: remove the verify_ownership parameter
         if verify_ownership {
```

Comment:
Maybe can modify in this pr?

if verify_ownership  maybe can modify to this:

```suggestion
        if !(db_name=="" && table_name=="") {
```

Reply:
Better to add next pr.